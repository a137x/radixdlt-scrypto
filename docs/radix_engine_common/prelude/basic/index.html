<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="SBOR basic, no custom types"><title>radix_engine_common::prelude::basic - Rust</title><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../static.files/SourceSerif4-Regular-46f98efaafac5295.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../static.files/FiraSans-Regular-018c141bf0843ffd.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../static.files/FiraSans-Medium-8f9a781e4970d388.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../static.files/SourceCodePro-Regular-562dcc5011b6de7d.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../static.files/SourceCodePro-Semibold-d899c5a5c4aeb14a.ttf.woff2"><link rel="stylesheet" href="../../../static.files/normalize-76eba96aa4d2e634.css"><link rel="stylesheet" href="../../../static.files/rustdoc-5bc39a1768837dd0.css"><meta name="rustdoc-vars" data-root-path="../../../" data-static-root-path="../../../static.files/" data-current-crate="radix_engine_common" data-themes="" data-resource-suffix="" data-rustdoc-version="1.77.1 (7cf61ebde 2024-03-27)" data-channel="1.77.1" data-search-js="search-dd67cee4cfa65049.js" data-settings-js="settings-4313503d2e1961c2.js" ><script src="../../../static.files/storage-4c98445ec4002617.js"></script><script defer src="../sidebar-items.js"></script><script defer src="../../../static.files/main-48f368f3872407c8.js"></script><noscript><link rel="stylesheet" href="../../../static.files/noscript-04d5337699b92874.css"></noscript><link rel="alternate icon" type="image/png" href="../../../static.files/favicon-16x16-8b506e7a72182f1c.png"><link rel="alternate icon" type="image/png" href="../../../static.files/favicon-32x32-422f7d1d52889060.png"><link rel="icon" type="image/svg+xml" href="../../../static.files/favicon-2c020d218678b618.svg"></head><body class="rustdoc mod"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle" title="show sidebar"></button></nav><nav class="sidebar"><div class="sidebar-crate"><h2><a href="../../../radix_engine_common/index.html">radix_engine_common</a><span class="version">1.1.2</span></h2></div><h2 class="location"><a href="#">Module basic</a></h2><div class="sidebar-elems"><section><ul class="block"><li><a href="#modules">Modules</a></li><li><a href="#enums">Enums</a></li><li><a href="#constants">Constants</a></li><li><a href="#traits">Traits</a></li><li><a href="#functions">Functions</a></li><li><a href="#types">Type Aliases</a></li></ul></section><h2><a href="../index.html">In radix_engine_common::prelude</a></h2></div></nav><div class="sidebar-resizer"></div>
    <main><div class="width-limiter"><nav class="sub"><form class="search-form"><span></span><div id="sidebar-button" tabindex="-1"><a href="../../../radix_engine_common/all.html" title="show sidebar"></a></div><input class="search-input" name="search" aria-label="Run search in the documentation" autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"><div id="help-button" tabindex="-1"><a href="../../../help.html" title="help">?</a></div><div id="settings-menu" tabindex="-1"><a href="../../../settings.html" title="settings"><img width="22" height="22" alt="Change settings" src="../../../static.files/wheel-7b819b6101059cd0.svg"></a></div></form></nav><section id="main-content" class="content"><div class="main-heading"><h1>Module <a href="../../index.html">radix_engine_common</a>::<wbr><a href="../index.html">prelude</a>::<wbr><a class="mod" href="#">basic</a><button id="copy-path" title="Copy item path to clipboard"><img src="../../../static.files/clipboard-7571035ce49a181d.svg" width="19" height="18" alt="Copy item path"></button></h1><span class="out-of-band"><a class="src" href="../../../src/sbor/lib.rs.html#9">source</a> · <button id="toggle-all-docs" title="collapse all docs">[<span>&#x2212;</span>]</button></span></div><details class="toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>SBOR basic, no custom types</p>
</div></details><h2 id="modules" class="section-header">Modules<a href="#modules" class="anchor">§</a></h2><ul class="item-table"><li><div class="item-name"><a class="mod" href="well_known_basic_custom_types/index.html" title="mod radix_engine_common::prelude::basic::well_known_basic_custom_types">well_known_basic_custom_types</a></div></li></ul><h2 id="enums" class="section-header">Enums<a href="#enums" class="anchor">§</a></h2><ul class="item-table"><li><div class="item-name"><a class="enum" href="enum.NoCustomExtension.html" title="enum radix_engine_common::prelude::basic::NoCustomExtension">NoCustomExtension</a></div></li><li><div class="item-name"><a class="enum" href="enum.NoCustomSchema.html" title="enum radix_engine_common::prelude::basic::NoCustomSchema">NoCustomSchema</a></div></li><li><div class="item-name"><a class="enum" href="enum.NoCustomTerminalValueRef.html" title="enum radix_engine_common::prelude::basic::NoCustomTerminalValueRef">NoCustomTerminalValueRef</a></div></li><li><div class="item-name"><a class="enum" href="enum.NoCustomTraversal.html" title="enum radix_engine_common::prelude::basic::NoCustomTraversal">NoCustomTraversal</a></div></li><li><div class="item-name"><a class="enum" href="enum.NoCustomTypeKind.html" title="enum radix_engine_common::prelude::basic::NoCustomTypeKind">NoCustomTypeKind</a></div></li><li><div class="item-name"><a class="enum" href="enum.NoCustomTypeValidation.html" title="enum radix_engine_common::prelude::basic::NoCustomTypeValidation">NoCustomTypeValidation</a></div></li><li><div class="item-name"><a class="enum" href="enum.NoCustomValue.html" title="enum radix_engine_common::prelude::basic::NoCustomValue">NoCustomValue</a></div></li><li><div class="item-name"><a class="enum" href="enum.NoCustomValueKind.html" title="enum radix_engine_common::prelude::basic::NoCustomValueKind">NoCustomValueKind</a></div></li></ul><h2 id="constants" class="section-header">Constants<a href="#constants" class="anchor">§</a></h2><ul class="item-table"><li><div class="item-name"><a class="constant" href="constant.BASIC_SBOR_V1_MAX_DEPTH.html" title="constant radix_engine_common::prelude::basic::BASIC_SBOR_V1_MAX_DEPTH">BASIC_SBOR_V1_MAX_DEPTH</a></div></li><li><div class="item-name"><a class="constant" href="constant.BASIC_SBOR_V1_PAYLOAD_PREFIX.html" title="constant radix_engine_common::prelude::basic::BASIC_SBOR_V1_PAYLOAD_PREFIX">BASIC_SBOR_V1_PAYLOAD_PREFIX</a></div></li></ul><h2 id="traits" class="section-header">Traits<a href="#traits" class="anchor">§</a></h2><ul class="item-table"><li><div class="item-name"><a class="trait" href="trait.BasicCategorize.html" title="trait radix_engine_common::prelude::basic::BasicCategorize">BasicCategorize</a></div></li><li><div class="item-name"><a class="trait" href="trait.BasicDecode.html" title="trait radix_engine_common::prelude::basic::BasicDecode">BasicDecode</a></div></li><li><div class="item-name"><a class="trait" href="trait.BasicDescribe.html" title="trait radix_engine_common::prelude::basic::BasicDescribe">BasicDescribe</a></div></li><li><div class="item-name"><a class="trait" href="trait.BasicEncode.html" title="trait radix_engine_common::prelude::basic::BasicEncode">BasicEncode</a></div></li><li><div class="item-name"><a class="trait" href="trait.BasicSbor.html" title="trait radix_engine_common::prelude::basic::BasicSbor">BasicSbor</a></div></li><li><div class="item-name"><a class="trait" href="trait.BasicSborEnum.html" title="trait radix_engine_common::prelude::basic::BasicSborEnum">BasicSborEnum</a></div></li><li><div class="item-name"><a class="trait" href="trait.BasicSborTuple.html" title="trait radix_engine_common::prelude::basic::BasicSborTuple">BasicSborTuple</a></div></li></ul><h2 id="functions" class="section-header">Functions<a href="#functions" class="anchor">§</a></h2><ul class="item-table"><li><div class="item-name"><a class="fn" href="fn.basic_decode.html" title="fn radix_engine_common::prelude::basic::basic_decode">basic_decode</a></div><div class="desc docblock-short">Decode an instance of <code>T</code> from a slice.</div></li><li><div class="item-name"><a class="fn" href="fn.basic_decode_with_depth_limit.html" title="fn radix_engine_common::prelude::basic::basic_decode_with_depth_limit">basic_decode_with_depth_limit</a></div></li><li><div class="item-name"><a class="fn" href="fn.basic_encode.html" title="fn radix_engine_common::prelude::basic::basic_encode">basic_encode</a></div><div class="desc docblock-short">Encode a <code>T</code> into byte array.</div></li><li><div class="item-name"><a class="fn" href="fn.basic_encode_with_depth_limit.html" title="fn radix_engine_common::prelude::basic::basic_encode_with_depth_limit">basic_encode_with_depth_limit</a></div></li><li><div class="item-name"><a class="fn" href="fn.basic_payload_traverser.html" title="fn radix_engine_common::prelude::basic::basic_payload_traverser">basic_payload_traverser</a></div><div class="desc docblock-short">Creates a payload traverser from the buffer</div></li></ul><h2 id="types" class="section-header">Type Aliases<a href="#types" class="anchor">§</a></h2><ul class="item-table"><li><div class="item-name"><a class="type" href="type.BasicDecoder.html" title="type radix_engine_common::prelude::basic::BasicDecoder">BasicDecoder</a></div></li><li><div class="item-name"><a class="type" href="type.BasicEncoder.html" title="type radix_engine_common::prelude::basic::BasicEncoder">BasicEncoder</a></div></li><li><div class="item-name"><a class="type" href="type.BasicOwnedRawPayload.html" title="type radix_engine_common::prelude::basic::BasicOwnedRawPayload">BasicOwnedRawPayload</a></div></li><li><div class="item-name"><a class="type" href="type.BasicOwnedRawValue.html" title="type radix_engine_common::prelude::basic::BasicOwnedRawValue">BasicOwnedRawValue</a></div></li><li><div class="item-name"><a class="type" href="type.BasicRawPayload.html" title="type radix_engine_common::prelude::basic::BasicRawPayload">BasicRawPayload</a></div></li><li><div class="item-name"><a class="type" href="type.BasicRawValue.html" title="type radix_engine_common::prelude::basic::BasicRawValue">BasicRawValue</a></div></li><li><div class="item-name"><a class="type" href="type.BasicSchema.html" title="type radix_engine_common::prelude::basic::BasicSchema">BasicSchema</a></div></li><li><div class="item-name"><a class="type" href="type.BasicTraverser.html" title="type radix_engine_common::prelude::basic::BasicTraverser">BasicTraverser</a></div></li><li><div class="item-name"><a class="type" href="type.BasicTypeData.html" title="type radix_engine_common::prelude::basic::BasicTypeData">BasicTypeData</a></div></li><li><div class="item-name"><a class="type" href="type.BasicTypeKind.html" title="type radix_engine_common::prelude::basic::BasicTypeKind">BasicTypeKind</a></div></li><li><div class="item-name"><a class="type" href="type.BasicValue.html" title="type radix_engine_common::prelude::basic::BasicValue">BasicValue</a></div></li><li><div class="item-name"><a class="type" href="type.BasicValueKind.html" title="type radix_engine_common::prelude::basic::BasicValueKind">BasicValueKind</a></div></li><li><div class="item-name"><a class="type" href="type.BasicVersionedSchema.html" title="type radix_engine_common::prelude::basic::BasicVersionedSchema">BasicVersionedSchema</a></div></li></ul></section></div></main></body></html>