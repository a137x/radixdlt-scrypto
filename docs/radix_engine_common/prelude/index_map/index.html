<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="The methods and macros provided directly in this `index_map` module (`new`, `with_capacity`) work in both std and no-std modes - unlike the corresponding methods on `IndexMap` itself."><title>radix_engine_common::prelude::index_map - Rust</title><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../static.files/SourceSerif4-Regular-46f98efaafac5295.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../static.files/FiraSans-Regular-018c141bf0843ffd.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../static.files/FiraSans-Medium-8f9a781e4970d388.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../static.files/SourceCodePro-Regular-562dcc5011b6de7d.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../static.files/SourceCodePro-Semibold-d899c5a5c4aeb14a.ttf.woff2"><link rel="stylesheet" href="../../../static.files/normalize-76eba96aa4d2e634.css"><link rel="stylesheet" href="../../../static.files/rustdoc-5bc39a1768837dd0.css"><meta name="rustdoc-vars" data-root-path="../../../" data-static-root-path="../../../static.files/" data-current-crate="radix_engine_common" data-themes="" data-resource-suffix="" data-rustdoc-version="1.77.1 (7cf61ebde 2024-03-27)" data-channel="1.77.1" data-search-js="search-dd67cee4cfa65049.js" data-settings-js="settings-4313503d2e1961c2.js" ><script src="../../../static.files/storage-4c98445ec4002617.js"></script><script defer src="../sidebar-items.js"></script><script defer src="../../../static.files/main-48f368f3872407c8.js"></script><noscript><link rel="stylesheet" href="../../../static.files/noscript-04d5337699b92874.css"></noscript><link rel="alternate icon" type="image/png" href="../../../static.files/favicon-16x16-8b506e7a72182f1c.png"><link rel="alternate icon" type="image/png" href="../../../static.files/favicon-32x32-422f7d1d52889060.png"><link rel="icon" type="image/svg+xml" href="../../../static.files/favicon-2c020d218678b618.svg"></head><body class="rustdoc mod"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle" title="show sidebar"></button></nav><nav class="sidebar"><div class="sidebar-crate"><h2><a href="../../../radix_engine_common/index.html">radix_engine_common</a><span class="version">1.1.2</span></h2></div><h2 class="location"><a href="#">Module index_map</a></h2><div class="sidebar-elems"><section><ul class="block"><li><a href="#macros">Macros</a></li><li><a href="#functions">Functions</a></li><li><a href="#types">Type Aliases</a></li></ul></section><h2><a href="../index.html">In radix_engine_common::prelude</a></h2></div></nav><div class="sidebar-resizer"></div>
    <main><div class="width-limiter"><nav class="sub"><form class="search-form"><span></span><div id="sidebar-button" tabindex="-1"><a href="../../../radix_engine_common/all.html" title="show sidebar"></a></div><input class="search-input" name="search" aria-label="Run search in the documentation" autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"><div id="help-button" tabindex="-1"><a href="../../../help.html" title="help">?</a></div><div id="settings-menu" tabindex="-1"><a href="../../../settings.html" title="settings"><img width="22" height="22" alt="Change settings" src="../../../static.files/wheel-7b819b6101059cd0.svg"></a></div></form></nav><section id="main-content" class="content"><div class="main-heading"><h1>Module <a href="../../index.html">radix_engine_common</a>::<wbr><a href="../index.html">prelude</a>::<wbr><a class="mod" href="#">index_map</a><button id="copy-path" title="Copy item path to clipboard"><img src="../../../static.files/clipboard-7571035ce49a181d.svg" width="19" height="18" alt="Copy item path"></button></h1><span class="out-of-band"><button id="toggle-all-docs" title="collapse all docs">[<span>&#x2212;</span>]</button></span></div><details class="toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>The methods and macros provided directly in this <code>index_map</code> module (<code>new</code>, <code>with_capacity</code>) work in both std and no-std modes - unlike the
corresponding methods on <code>IndexMap</code> itself.</p>
<p>Unfortunately <code>IndexMap</code> is very hard to use from no-std (see <a href="https://docs.rs/indexmap/latest/indexmap/#no-standard-library-targets">docs</a>
and <a href="https://github.com/bluss/indexmap/issues/184">relevant github issue</a>). It uses a weird build flag to detect if no-std is present, which
is hard to force unless you explicitly do eg a WASM build and see that it’s missing.</p>
<p>The recommended way to use IndexMap is to add <code>use sbor::rust::collections::*</code> and then reference the type inline as <code>index_map::IndexMap</code>
and create new sets using <code>index_map::new</code>, <code>index_map::with_capacity</code>, or the <code>index_map::indexmap!</code> macro. Always putting the <code>index_map</code>
mod will help enforce the use of these methods instead of the methods on <code>IndexMap</code> itself.</p>
<p>You can use these exports as follows:</p>

<div class="example-wrap"><pre class="rust rust-example-rendered"><code><span class="kw">use </span>sbor::rust::collections::<span class="kw-2">*</span>;

<span class="kw">let </span>index_map: IndexMap&lt;K, V&gt; = index_map_new();
<span class="kw">let </span>index_map: IndexMap&lt;K, V&gt; = index_map_with_capacity(n);
<span class="kw">let </span>index_map = <span class="macro">indexmap!</span>(<span class="number">1u32 </span>=&gt; <span class="string">"entry_one"</span>, <span class="number">5u32 </span>=&gt; <span class="string">"entry_two"</span>);</code></pre></div>
</div></details><h2 id="macros" class="section-header">Macros<a href="#macros" class="anchor">§</a></h2><ul class="item-table"><li><div class="item-name"><a class="macro" href="macro.indexmap.html" title="macro radix_engine_common::prelude::index_map::indexmap">indexmap</a></div></li></ul><h2 id="functions" class="section-header">Functions<a href="#functions" class="anchor">§</a></h2><ul class="item-table"><li><div class="item-name"><a class="fn" href="fn.new.html" title="fn radix_engine_common::prelude::index_map::new">new</a></div><div class="desc docblock-short">This is safe for std and no-std use cases (unlike <code>IndexMap::new</code> which disappears when std is not in the toolchain - see
<a href="https://faultlore.com/blah/defaults-affect-inference/">this article</a> for deep technical reasons)</div></li><li><div class="item-name"><a class="fn" href="fn.with_capacity.html" title="fn radix_engine_common::prelude::index_map::with_capacity">with_capacity</a></div><div class="desc docblock-short">This is safe for std and no-std use cases (unlike <code>IndexMap::with_capacity</code> which disappears when std is not in the toolchain - see
<a href="https://faultlore.com/blah/defaults-affect-inference/">this article</a> for deep technical reasons)</div></li></ul><h2 id="types" class="section-header">Type Aliases<a href="#types" class="anchor">§</a></h2><ul class="item-table"><li><div class="item-name"><a class="type" href="type.DefaultHashBuilder.html" title="type radix_engine_common::prelude::index_map::DefaultHashBuilder">DefaultHashBuilder</a></div></li><li><div class="item-name"><a class="type" href="type.IndexMap.html" title="type radix_engine_common::prelude::index_map::IndexMap">IndexMap</a></div></li></ul></section></div></main></body></html>