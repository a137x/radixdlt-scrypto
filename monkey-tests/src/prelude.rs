pub use radix_engine_common::prelude::*;
pub use radix_engine_interface::prelude::*;
pub use transaction::builder::*;
