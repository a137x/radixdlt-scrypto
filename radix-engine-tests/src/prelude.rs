#![allow(ambiguous_glob_reexports)]

pub use crate::common::*;
pub use native_sdk::account::*;
pub use native_sdk::component::*;
pub use native_sdk::modules::*;
pub use native_sdk::resource::*;
pub use paste::paste;
pub use radix_engine::blueprints::access_controller::*;
pub use radix_engine::blueprints::account::*;
pub use radix_engine::blueprints::consensus_manager::*;
pub use radix_engine::blueprints::identity::*;
pub use radix_engine::blueprints::models::*;
pub use radix_engine::blueprints::native_schema::*;
pub use radix_engine::blueprints::package::*;
pub use radix_engine::blueprints::pool::v1::constants::*;
pub use radix_engine::blueprints::pool::v1::errors::multi_resource_pool::Error as MultiResourcePoolError;
pub use radix_engine::blueprints::pool::v1::errors::one_resource_pool::Error as OneResourcePoolError;
pub use radix_engine::blueprints::pool::v1::errors::two_resource_pool::Error as TwoResourcePoolError;
pub use radix_engine::blueprints::pool::*;
pub use radix_engine::blueprints::resource::*;
pub use radix_engine::blueprints::transaction_processor::*;
pub use radix_engine::blueprints::transaction_tracker::*;
pub use radix_engine::blueprints::util::*;
pub use radix_engine::errors::*;
pub use radix_engine::kernel::call_frame::*;
pub use radix_engine::kernel::heap::*;
pub use radix_engine::kernel::id_allocator::*;
pub use radix_engine::kernel::kernel::*;
pub use radix_engine::kernel::kernel_api::*;
pub use radix_engine::kernel::kernel_callback_api::*;
pub use radix_engine::kernel::substate_io::*;
pub use radix_engine::kernel::substate_locks::*;
pub use radix_engine::system::actor::*;
pub use radix_engine::system::bootstrap::*;
pub use radix_engine::system::checkers::*;
pub use radix_engine::system::system::*;
pub use radix_engine::system::system_callback::*;
pub use radix_engine::system::system_callback_api::*;
pub use radix_engine::system::system_db_reader::*;
pub use radix_engine::system::system_modules::auth::*;
pub use radix_engine::system::system_modules::costing::*;
pub use radix_engine::system::system_modules::execution_trace::*;
pub use radix_engine::system::system_modules::limits::*;
pub use radix_engine::system::system_modules::*;
pub use radix_engine::system::system_substates::*;
pub use radix_engine::system::system_type_checker::*;
pub use radix_engine::track::*;
pub use radix_engine::transaction::*;
pub use radix_engine::utils::*;
pub use radix_engine::vm::wasm::*;
pub use radix_engine::vm::wasm_runtime::*;
pub use radix_engine::vm::*;
pub use radix_engine_common::math::*;
pub use radix_engine_common::prelude::*;
pub use radix_engine_interface::api::actor_api::*;
pub use radix_engine_interface::api::actor_index_api::*;
pub use radix_engine_interface::api::actor_key_value_entry_api::*;
pub use radix_engine_interface::api::actor_sorted_index_api::*;
pub use radix_engine_interface::api::blueprint_api::*;
pub use radix_engine_interface::api::field_api::*;
pub use radix_engine_interface::api::key_value_entry_api::*;
pub use radix_engine_interface::api::key_value_store_api::*;
pub use radix_engine_interface::api::node_modules::*;
pub use radix_engine_interface::api::object_api::*;
pub use radix_engine_interface::api::system_modules::transaction_runtime_api::*;
pub use radix_engine_interface::api::system_modules::*;
pub use radix_engine_interface::api::*;
pub use radix_engine_interface::blueprints::access_controller::*;
pub use radix_engine_interface::blueprints::account::*;
pub use radix_engine_interface::blueprints::consensus_manager::*;
pub use radix_engine_interface::blueprints::identity::*;
pub use radix_engine_interface::blueprints::pool::*;
pub use radix_engine_interface::blueprints::resource::*;
pub use radix_engine_interface::blueprints::test_utils::invocations::*;
pub use radix_engine_interface::blueprints::transaction_processor::*;
pub use radix_engine_interface::prelude::*;
pub use radix_engine_queries::typed_native_events::*;
pub use radix_engine_queries::typed_substate_layout::*;
pub use radix_engine_store_interface::db_key_mapper::*;
pub use radix_engine_store_interface::interface::*;
pub use radix_engine_stores::memory_db::*;
pub use rand::prelude::*;
pub use rand::*;
pub use rand_chacha::ChaCha8Rng;
pub use sbor::basic_well_known_types::*;
pub use scrypto::prelude::*;
pub use scrypto_test::environment::*;
pub use scrypto_test::sdk::*;
pub use scrypto_unit::*;
pub use transaction::builder::*;
pub use transaction::errors::*;
pub use transaction::manifest::e2e::*;
pub use transaction::manifest::*;
pub use transaction::model::*;
pub use transaction::validation::*;
pub use transaction_scenarios::scenario::*;
pub use transaction_scenarios::scenarios::*;
pub use walkdir::WalkDir;

#[cfg(feature = "std")]
pub use only_std::*;

#[cfg(feature = "std")]
mod only_std {
    pub use std::fs::File;
}
