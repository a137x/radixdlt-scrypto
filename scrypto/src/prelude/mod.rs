//=============
// This crate
//=============

pub use crate::component::*;
pub use crate::crypto_utils::*;
pub use crate::engine::scrypto_env::ScryptoVmV1Api;
pub use crate::engine::*;
pub use crate::modules::*;
pub use crate::resource::*;
pub use crate::runtime::*;
pub use crate::{
    blueprint, component_royalties, component_royalty_config, debug, enable_function_auth,
    enable_method_auth, enable_package_royalties, error, extern_blueprint_internal, include_code,
    include_schema, info, internal_add_role, internal_component_royalty_entry, main_accessibility,
    method_accessibilities, method_accessibility, role_list, roles, this_package, to_role_key,
    trace, warn, NonFungibleData,
};

//=========================
// Radix Engine Interface
//=========================

pub use module_blueprints_interface::auth::*;
pub use module_blueprints_interface::metadata::*;
pub use module_blueprints_interface::*;
pub use native_blueprints_interface::consensus_manager::TimePrecision;
pub use native_blueprints_interface::resource::*;
pub use num_traits::{
    cast::FromPrimitive, cast::ToPrimitive, identities::One, identities::Zero, pow::Pow,
    sign::Signed,
};
pub use radix_engine_common::crypto::*;
pub use radix_engine_common::math::*;
pub use radix_engine_common::prelude::*;
pub use radix_engine_common::{
    access_and_or, access_rule_node, i, internal_roles, manifest_args, metadata, metadata_init,
    metadata_init_set_entry, role_entry, roles2, rule, scrypto_args, ScryptoCategorize,
    ScryptoDecode, ScryptoEncode, ScryptoEvent, ScryptoSbor,
};

//=======
// SBOR
//=======

pub use sbor::{Categorize, Decode, DecodeError, Encode, Sbor};

// Needed for macros
pub use radix_engine_common::radix_engine_common;

/// We should always `UncheckedUrl` in Scrypto, as the validation logic is heavy.
/// Thus, this type alias is added.
pub type Url = UncheckedUrl;

/// We should always `UncheckedOrigin` in Scrypto, as the validation logic is heavy.
/// Thus, this type alias is added.
pub type Origin = UncheckedOrigin;
