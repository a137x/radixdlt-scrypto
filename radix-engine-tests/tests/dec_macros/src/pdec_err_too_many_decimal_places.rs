use radix_engine_interface::prelude::*;

fn main() {
    const X: PreciseDecimal = pdec!(1.0000000000000000000000000000000000001);
}
