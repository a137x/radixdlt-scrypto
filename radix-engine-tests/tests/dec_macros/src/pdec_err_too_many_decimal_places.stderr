error: A precise decimal cannot have more than thirty-six decimal places.
 --> src/pdec_err_too_many_decimal_places.rs:4:37
  |
4 |     const X: PreciseDecimal = pdec!(1.0000000000000000000000000000000000001);
  |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
